zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -808,15 +809,17 @@ public int compare(PathWithStat left, PathWithStat right) {
      * Produces a paginated list of the children of a given path
      * @param path path of node node to list
      * @param stat stat of the node to list
-     * @param watcher an optional watcher to attach to the node. The watcher is added only once when reaching the end of pagination
+     * @param watcher an optional watcher to attach to the node. The watcher is added only once when reaching the end of
+     *               pagination
      * @param maxReturned maximum number of children to return. Return one more than this number to indicate truncation
      * @param minCzxId only return children whose creation zxid equal or greater than minCzxId
      * @param czxIdOffset how many children with zxid == minCzxId to skip (as returned in previous pages)
-     * @return A list of path with stats
+     * @param nextPage info to be used for the next page call
+     * @return A list of child names of the given path
      * @throws NoNodeException if the path does not exist
      */
-    public List<PathWithStat> getPaginatedChildren(String path, Stat stat, Watcher watcher, int maxReturned,
-                                                   long minCzxId, long czxIdOffset)
+    public List<String> getPaginatedChildren(String path, Stat stat, Watcher watcher, int maxReturned,
+                                             long minCzxId, long czxIdOffset, PaginationNextPage nextPage)
```

Comment:
Where is it updated?

Reply:
Updated in `DataTree`. Now this PR should have the complete logic.